share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+3. To know about various wake options run `wake --help`. `v` stands for verbosity and `x` stands for execute the expression.
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run them using `wake` command in the terminal.  
+
+5. For more information on wake library click [here](https://sifive.github.io/wake/)
+
+### wake `Hello World`
+* If you want to display a string you can do that using the below function by writing the string to be displayed inside "" and call the function abc . Ex: `wake -x 'abc'`. This will call the function `abc` and prints `Hello World`. 
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defined using the keyword `def`. Simple function definition takes the form
+`def <function name> <parameter list> = <expression>`
+
```

Comment:
are there types? Can you clarify right away if there is any way to define the type of arguments or return values?

Reply:
also, is there anything *other* than `def`? Is there no `val`? That might be an important thing to state up front: "the only thing you can define are functions using `def` keyword".

Reply:
There is no `val`. We define everything using `def`

Reply:
There is no way to specify the type of arguments. In a multi-line function body return value is specified by the last line in the function